### PR TITLE
Cleanup cluster if error during creation

### PR DESCRIPTION
**Pull request by leahecole**
## Description

We currently are having problems with dataproc clusters in our test environment (see b/211687420 if you're internal) and I figured out that these tests had leftovers if the dataproc fixtures failed which meant we were hitting quotas. This cleans up the leftovers if theres an error creating the cluster and still raises the error.

Note: It's a good idea to open an issue first for discussion.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

**Comment by leahecole** [comment]
Snippet bot is only triggered because I ran blacken on some region tags

**Comment by leahecole** [comment]
Also, tests will likely error with the inability to create the dataproc cluster. That's expected right now and okay - the great part about this PR is that no matter how many times we run the tests it won't leave cluster leftovers 😁

File: data-science-onramp/data-ingestion/ingestion_test.py [hunk 1]
```diff
@@ -73,25 +74,38 @@
...
-    )
-    operation.result()
+    try:
+        # Delete cluster
+        operation = cluster_client.delete_cluster(
+            project_id=PROJECT_ID, region=CLUSTER_REGION, cluster_name=DATAPROC_CLUSTER
+        )
+        operation.result()
+    except NotFound:
+        print("Cluster already deleted")
```
Comment (tswast) [inline data-science-onramp/data-ingestion/ingestion_test.py:108]: Thoughts on moving `yield` to inside the `try:` block and using `finally` instead?

```python
        # Wait for cluster to provision
        operation.result()
        yield
    finally:
        # Clean up leftovers if there is an error creating the fixture
        try:
            operation = cluster_client.delete_cluster(
                project_id=PROJECT_ID, region=CLUSTER_REGION, cluster_name=DATAPROC_CLUSTER
            )
            operation.result()
        except NotFound:
            print("Cluster already deleted")
```

**Review by tswast — COMMENTED** [review]
Looks like we might still be having some networking issues post-log4j fallout?

> google.api_core.exceptions.InternalServerError: 500 Cannot start master: Timed out waiting for 2 nodes. This usually happens when VM to VM communications are blocked by firewall rules.

File: data-science-onramp/data-ingestion/ingestion_test.py (hunk 1 above)
Comment (leahecole) [reply data-science-onramp/data-ingestion/ingestion_test.py:108]: great idea! sometimes I forget about finally. Thanks for the reminder - I'll make that change now!

**Review by tswast — APPROVED** [review]

**Review by busunkim96 — APPROVED** [review]

**Pull request merged**